WordPress/gutenberg
# Add new block commands

## What?


Advances: #51559
Adds block selection related commands: block transforms and block duplicate, copy, remove, insert after/before etc.

## Why?


Mainly to improve the usefulness of the command center, but also because this improve the distraction free mode experience by offering immediate access to basic functions without turn off the mode.

## How?


Register the commands using mostly copied functionality from the block switcher and block actions components.

## Testing Instructions

1. Select some block
2. Press cmd + k 
3. Type "transform to" or "copy"
4. See command
5. Execute command
6. It should do what it says

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/107534/b02d9053-41db-4816-8178-3299fbb3d61b

## To do

- [x] Does this make things slower? I notice some lagginess but it may be just me.
- [x] Cleanup some of the copy pasta (there's mentions of patterns there).
- [x] Is the block list the best place to register these commands?
- [x] Should these commands use a context?
